web/src/features/map/Map.tsx

```diff
@@ -102,11 +112,15 @@ export default function MapPage(): ReactElement {
     }
 
     // An issue where the map has not loaded source yet causing map errors
-    const isSourceLoaded = map.getSource('zones-clickable') != undefined;
+    const isSourceLoaded =
+      map.getSource('zones-clickable') != undefined &&
+      map.isSourceLoaded('zones-clickable') &&
+      map.isSourceLoaded('states') &&
+      map.getSource('states') != undefined;
```

Comment:
Should the order of 'states' here be opposite? E.g. first check that it is not undefined, then check if it is loaded